testdata/p4_16_samples_outputs/issue513-midend.p4

```diff
@@ -0,0 +1,68 @@
+#include <core.p4>
+#define V1MODEL_VERSION 20180101
+#include <v1model.p4>
+
+header Ethernet_h {
+    bit<48> dstAddr;
+    bit<48> srcAddr;
+    bit<16> etherType;
+}
+
+struct Parsed_packet {
+    Ethernet_h ethernet;
+}
+
+struct mystruct1 {
+    bit<4> a;
+    bit<4> b;
+}
+
+control DeparserI(packet_out packet, in Parsed_packet hdr) {
+    apply {
+        packet.emit<Ethernet_h>(hdr.ethernet);
+    }
+}
+
+parser parserI(packet_in pkt, out Parsed_packet hdr, inout mystruct1 meta, inout standard_metadata_t stdmeta) {
+    state start {
+        pkt.extract<Ethernet_h>(hdr.ethernet);
+        transition accept;
+    }
+}
+
+control cIngress(inout Parsed_packet hdr, inout mystruct1 meta, inout standard_metadata_t stdmeta) {
+    @name("cIngress.foo") action foo() {
+        meta.b = meta.b + 4w5;
+        mark_to_drop();
```

Comment:
There is no 'if' statement in this action, which appears to be incorrect, as it does not match the behavior of the input program.

Reply:
maybe it's not that easy, I guess the pass was not designed for this case.
I will mark this as draft for now.

Reply:
If you expect this might not happen quickly, Mihai, would you (or anyone reading this) be able to provide the p4test command line option that I asked about in this p4-arch email? https://groups.google.com/a/lists.p4.org/g/p4-arch/c/FD-IxwekCUI/m/9hSFxPSWAwAJ

Reply:
> can you recommend a full p4test command line that would skip the fewest passes possible, and would not give warnings for 'if' conditions inside of action bodies?

Hi Andy, as far as I understand, the warning for `if` conditions inside of action bodies happens at the _Predication_ midend pass.
https://github.com/p4lang/p4c/blob/ceeb043ac19208ea34ec94fd1a1ddae7ee9e7ed1/midend/predication.h#L117-L119

Thus, `p4test --excludeMidendPasses Predication <program>.p4` command should not show this warning. For example:
```sh
p4test --excludeMidendPasses Predication testdata/p4_16_samples/pna-example-tcp-connection-tracking.p4
```

Reply:
@rst0git That worked for me.  Thanks, Radostin!